### ISSUE TITLE
cve-2018-14647  and Heketi image

### ISSUE THREAD

**Issue by chandidas** [heketi/heketi#1550]
### Kind of issue

> Feature request

**Details:**
python-cve201814647-dos (150579) reported Sep 26, 2018

Python is vulnerable to a denial of service, caused by a flaw in the elementtree C accelerator. By using a specially-crafted XML document, a remote attacker could exploit this vulnerability to cause a resource exhaustion.

Python 3.8, 3.7, 3.6, 3.5, 3.4, 2.7 are believed to be vulnerable

**Consequences:**
Denial of Service

**Remedy:**
Per https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2018-14647, fixed in versions:
python 3.7.1, python 3.6.7, python 2.7.16

### Expected/desired behavior
When we create a Heketi image from CentOS as base image, python version is `2.7.5` which comes from base CentOS. 
Do community has any plan to upgrade python version to `2.7.16` to address the vulnerability reported.

**Comment by nixpanic** [comment]
This is more of a CentOS question than a Heketi one. We will include a fix for this when CentOS provides it. Which basically means it needs to get included in RHEL.

Because the impact is moderate (according to https://access.redhat.com/security/cve/cve-2018-14647), and no update for Python has been provided yet, I do not expect a fix for this before the next minor version of RHEL-7 is released.

**Comment by chandidas** [comment]
@nixpanic Thanks for your quick reply.
I was thinking if, instead of relying on base CentOS, latest python can be installed while creating heketi docker image itself. Do you see that as a possibility or is it like go with only the proven/tested version of python only.

**Comment by nixpanic** [comment]
No, I do not think we want to  do that. Red Hat normally backports
security fixes to the version that is part of their product. We should
not replace Python in the images.

Also note that Heketi does not use Python itself. Python is part of the
container because some components of RHEL/CentOS require it. The
security issue in Python is not exposed through the API that Heketi
offers.

**Comment by phlogistonjohn** [comment]
Hi @chandidas can we close this issue? I think @nixpanic wrote a fairly explanatory comment. We don't maintain the content of the base image we only consume it.

**Comment by chandidas** [comment]
Yes we can close this issue as @nixpanic has clarified it clearly. Thanks.
